tests/app/clients/test_firetext.py

```diff
@@ -8,19 +8,25 @@
 from app.clients.sms.firetext import get_firetext_responses, SmsClientResponseException, FiretextClientResponseException
 
 
-def test_should_return_correct_details_for_delivery():
-    get_firetext_responses('0') == 'delivered'
+@pytest.mark.parametrize('code, result', [
+    (None, ('delivered', None)), ('000', ('delivered', None))
+])
+def test_get_firetext_responses_should_return_correct_details_for_delivery(code, result):
+    assert get_firetext_responses('0', code) == result
 
 
-def test_should_return_correct_details_for_bounced():
-    get_firetext_responses('1') == 'permanent-failure'
+@pytest.mark.parametrize('code, result', [
+    (None, ('permanent-failure', None)), ('401', ('permanent-failure', 'Message Rejected'))
+])
+def test_get_firetext_responses_should_return_correct_details_for_bounced(code, result):
+    assert get_firetext_responses('1', code) == result
 
 
-def test_should_return_correct_details_for_complaint():
-    get_firetext_responses('2') == 'pending'
+def test_get_firetext_responses_should_return_correct_details_for_complaint():
+    assert get_firetext_responses('2') == ('pending', None)
 
 
-def test_should_be_none_if_unrecognised_status_code():
+def test_get_firetext_responses_should_be_none_if_unrecognised_status_code():
```

Comment:
Option: Rename test to say a be a bit clearer about what it is returning ie `test_get_firetext_responses_should_raise_error_if_unrecognised_status_code`